File: app/src/main/java/net/ducksmanager/persistence/models/composite/CoverSearchIssueWithUserIssueDetails.java
package net.ducksmanager.persistence.models.composite;

import net.ducksmanager.persistence.models.dm.Issue;

import androidx.room.Embedded;

public class CoverSearchIssueWithUserIssueDetails {
    @Embedded
    private final CoverSearchIssue coverSearchIssue;

    @Embedded
    private final Issue userIssue;

    public CoverSearchIssueWithUserIssueDetails(CoverSearchIssue coverSearchIssue, Issue userIssue) {
        this.coverSearchIssue = coverSearchIssue;
        this.userIssue = userIssue;
    }

    public CoverSearchIssue getCoverSearchIssue() {
        return coverSearchIssue;
    }

    public Issue getUserIssue() {
        return userIssue;
    }

}

File: app/src/main/java/net/ducksmanager/persistence/models/composite/CoverSearchIssueWithUserIssueDetails.kt
package net.ducksmanager.persistence.models.composite

import androidx.room.Embedded
import net.ducksmanager.persistence.models.dm.Issue

class CoverSearchIssueWithUserIssueDetails(
    @Embedded
    val coverSearchIssue: CoverSearchIssue,

    @Embedded
    val userIssue: Issue?
)